tests/e2e/workloads/test_pgsql_reset_app_pod.py

```diff
@@ -0,0 +1,136 @@
+"""
+Module to perform PGSQL workload
+"""
+import logging
+import pytest
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.utility import templating, utils
+from ocs_ci.utility.utils import run_cmd, TimeoutSampler
+from ocs_ci.ocs.utils import get_pod_name_by_pattern
+from ocs_ci.ocs.ripsaw import RipSaw
+from ocs_ci.ocs import constants
+from tests import disruption_helpers
+from ocs_ci.framework.testlib import (
+    E2ETest, google_api_required, workloads, ignore_leftovers
+)
+from ocs_ci.ocs.exceptions import UnexpectedBehaviour
+from ocs_ci.utility.spreadsheet.spreadsheet_api import GoogleSpreadSheetAPI
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope='function')
+def ripsaw(request):
+
+    # Create RipSaw Operator
+    ripsaw = RipSaw()
+
+    def teardown():
+        ripsaw.cleanup()
+    request.addfinalizer(teardown)
+    return ripsaw
+
+
+
+@workloads
+class TestPgSQLAppPodRespin(E2ETest):
+    """
+    Test running PGSQL and with Ceph pods respin
+    """
+    @pytest.fixture()
+    def pgsql_setup(self, ripsaw):
+        """
+        PGSQL test setup
+        """
+        # Deployment ripsaw and postgres database
+        log.info("Deploying postgres database")
+        ripsaw.apply_crd(
+            'resources/crds/'
+            'ripsaw_v1alpha1_ripsaw_crd.yaml'
+        )
+        ripsaw.setup_postgresql()
+
+    @pytest.mark.parametrize(
+        argnames=[
+            "transactions", "pod_name"
+        ],
+        argvalues=[
+            pytest.param(
+                *[600, 'postgres'], marks=pytest.mark.polarion_id("OCS-809")
+            )
+        ]
+    )
+    @pytest.mark.usefixtures(pgsql_setup.__name__)
+    def test_reset_app_pod_pgsql(self, transactions, pod_name):
```

Comment:
a lot of similarities to test_pgsql.py. Can we please consolidate this test function under test_pgsql.py's test class and avoid core duplicity?

Reply:
+1 , we discussed about this, Prasad is going to discuss with Oded to fit into one of the existing mon/osd/mgr failure test.